surfactant/infoextractors/docker_file.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2023 Lawrence Livermore National Security, LLC
+# See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: MIT
+import gzip
+import json
+import subprocess
+import tempfile
+
+import surfactant.plugin
+from surfactant.sbomtypes import SBOM, Software
+
+
+def supports_file(filetype: str) -> bool:
+    return filetype in ("Docker archive tar", "Docker archive gzip")
+
+
+@surfactant.plugin.hookimpl
+def extract_file_info(sbom: SBOM, software: Software, filename: str, filetype: str) -> object:
+    if not supports_file(filetype):
+        return None
+    return extract_docker_info(filetype, filename)
+
+
+def extract_docker_info(filetype: str, filename: str) -> object:
+    if filetype == "Docker archive gzip":
+        with open(filename, "rb") as gzip_in:
+            gzip_data = gzip_in.read()
+        with tempfile.NamedTemporaryFile() as gzip_out:
+            gzip_out.write(gzip.decompress(gzip_data))
+            return run_docker(gzip_out.name)
+    return run_docker(filename)
+
+
+# Function that extract_docker_info delegates to to actually run Docker scout
+def run_docker(filename: str) -> object:
+    result = subprocess.run(
+        ["docker", "scout", "sbom", "--format", "spdx", f"fs://{filename}"],
```

Comment:
`docker scout cves` requires being logged into Docker's website and may also require uploading the docker image (the documentation isn't very clear on this - it kind of just assumes you're using remote images)

Reply:
Looking at the [docs](https://docs.docker.com/reference/cli/docker/scout/cves/), it seems to have it run on something not uploaded to DockerHub requires adding a prefix -- `archive://` would probably be it for the docker tar files we are looking at.

If it gets run on a local file, does it still require being logged into Docker? Maybe this is something that would need to be a user configurable setting that defaults to off -- not really sure why it would need logging in though, other than if Docker is maintaining their own CVE/advisory database instead of using the public databases.